homeassistant/components/sensor/coinbase.py

```diff
@@ -0,0 +1,150 @@
+"""
+Support for Coinbase sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.coinbase/
+"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import ATTR_ATTRIBUTION
+
+
+DEPENDENCIES = ['coinbase']
+
+DATA_COINBASE = 'coinbase_cache'
+
+CONF_ATTRIBUTION = "Data provided by coinbase.com"
+ATTR_NATIVE_BALANCE = "Balance in native currency"
+
+BTC_ICON = 'mdi:currency-btc'
+ETH_ICON = 'mdi:currency-eth'
+COIN_ICON = 'mdi:coin'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Coinbase sensors."""
+    if discovery_info is None:
+        return False
+    if 'account' in discovery_info:
+        account = discovery_info['account']
+        sensor = CoinbaseAccountSensor(hass,
+                                       account['name'],
+                                       account['balance']['amount'],
+                                       account['balance']['currency'],
+                                       account['native_balance']['amount'],
+                                       account['native_balance']['currency'])
+    if 'exchange_currency' in discovery_info:
+        exchange_currency = discovery_info['exchange_currency']
+        native_currency = discovery_info['native_currency']
+        rate = discovery_info['rate']
+        sensor = CoinbaseExchangeRateSensor(hass,
+                                            exchange_currency,
+                                            rate,
+                                            native_currency)
+
+    add_devices([sensor], True)
+
+
+class CoinbaseAccountSensor(Entity):
+    """Representation of a Coinbase.com sensor."""
+
+    def __init__(self, hass, name, balance, currency, native_bal, native_cur):
```

Comment:
Don't pass in `hass`. Since you call update before adding the entity, I'd only pass in the coinbase data instance, name and currency here.